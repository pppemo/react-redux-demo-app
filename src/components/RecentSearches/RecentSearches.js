import React from 'react'
import { Label } from 'react-bootstrap'
import PropTypes from 'prop-types'
import styles from './RecentSearches.module.scss'

const RecentSearches = ({ label, searches, onSelected }) => (
  <div>
    <span>{label}{label && ': '}</span>
    {searches && searches.map(element => (
      <span key={element.code}>
        <h4 className={styles.labelSizer}>
          <Label className={styles.label}
            onClick={() => onSelected && onSelected(element.code)}>
            {element.name}
          </Label>
        </h4>
      </span>
    ))}
  </div>
)

RecentSearches.propTypes = {
  label: PropTypes.string,
  searches: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      code: PropTypes.string
    })
  ),
  onSelected: PropTypes.func,
  onClose: PropTypes.func
}

export default RecentSearches
