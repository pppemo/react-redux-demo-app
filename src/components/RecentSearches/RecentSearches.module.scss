.label {
  cursor: pointer;
}

.labelSizer {
  display: inline-block;
  margin: 0 5px;
}
