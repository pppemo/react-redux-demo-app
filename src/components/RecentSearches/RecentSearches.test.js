import TestRenderer from 'react-test-renderer'
import React from 'react'
import { shallow } from 'enzyme'
import RecentSearches from './RecentSearches'
import styles from './RecentSearches.module.scss'

const searches = [{
  name: 'Poland',
  code: 'PL'
}, {
  name: 'Germany',
  code: 'DE'
}]

describe('RecentSearches component', () => {
  it('should render with recent searches', () => {
    const tree = TestRenderer
      .create(<RecentSearches searches={searches}
        label="Component label" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render without recent searches', () => {
    const tree = TestRenderer
      .create(<RecentSearches label="Component label" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should trigger search', () => {
    const onSelected = jest.fn()
    const wrapper = shallow(<RecentSearches searches={searches}
      onSelected={onSelected} />)
    wrapper.find(`.${styles.label}`).first().simulate('click')
    expect(onSelected).toBeCalled()
  })
})
