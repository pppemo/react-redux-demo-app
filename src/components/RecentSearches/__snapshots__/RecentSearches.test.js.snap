// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentSearches component should render with recent searches 1`] = `
<div>
  <span>
    Component label
    : 
  </span>
  <span>
    <h4
      className="labelSizer"
    >
      <span
        className="label label label-default"
        onClick={[Function]}
      >
        Poland
      </span>
    </h4>
  </span>
  <span>
    <h4
      className="labelSizer"
    >
      <span
        className="label label label-default"
        onClick={[Function]}
      >
        Germany
      </span>
    </h4>
  </span>
</div>
`;

exports[`RecentSearches component should render without recent searches 1`] = `
<div>
  <span>
    Component label
    : 
  </span>
</div>
`;
